ci/check-each-commit.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/sh
+set -e
+set -x
+
+for REMOTE in `git remote`; do
+	if [ "$(git remote show $REMOTE | grep "^  Fetch URL: .*github.com\(:\|/\)rust-bitcoin/rust-lightning")" != "" ]; then
+		UPSTREAM=$REMOTE
+		break
+	fi
+done
+if [ "$UPSTREAM" = "" ]; then
+	echo "Failed to find a remote set to github.com/rust-bitcoin/rust-lightning"
+	exit 1
+fi
+if [ "$(git log --pretty="%H %D" | grep "^[0-9a-f]*.* $UPSTREAM/main")" = "" ]; then
+	echo "It seems like the current checked-out commit is not based on the latest usptream commit"
+	exit 1
+fi
+git rebase $UPSTREAM/main
+for COMMITHASH in `git log --format=format:%H $UPSTREAM/main...HEAD`; do
+	git checkout $COMMITHASH
+	cargo check
+	cd fuzz && cargo check --features=stdin_fuzz && cd ..
```

Comment:
Making fuzz opt-in with a flag might be a good idea to make this easy on developers. This gets into longer-than-a-coffee-break territory with just a few patches in a stack.

Reply:
Hmm, it really shouldnt - cargo check does some caching, so while the first commit might take a minute, the later ones should all be pretty much instant. More importantly, breaking fuzz compile is one of the most common issues, so it'd be kinda bad to default it off.

Reply:
Ah yes. I read this as running the fuzz tests. I completely agree that these should be built.